### ISSUE TITLE
FALSE NEGATIVE: `UL_UNRELEASED_LOCK` misses methods with no `unlock()` call

### ISSUE THREAD

**Issue by rikki322** [spotbugs/spotbugs#4310]
Hi, I found a false negative in SpotBugs 4.10.3 when a method acquires a lock and contains no `unlock()` call.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs UL_UNRELEASED_LOCK`

**Minimal reproducer**

```java
import java.util.concurrent.locks.Lock;
import java.util.concurrent.locks.ReentrantLock;

class SpotbugsSonarqubeNoUnlockCall {
  private final Lock lock = new ReentrantLock();

  void test() {
    lock.lock();
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-sonarqube-ul-unreleased-lock-fn-no-unlock-call.java
spotbugs -textui -effort:max -low -visitors FindUnreleasedLock classes
```

**Current behavior**

SpotBugs produces no `UL_UNRELEASED_LOCK` diagnostic.

**Expected behavior**

SpotBugs should report line 8 because the method acquires `lock` and never releases it.
